cert/cert.go

```diff
@@ -398,34 +434,34 @@ func (spec *Spec) Lifespan() time.Duration {
 	}
 
 	// This bit of code is necessary to confirm that the cert/key are older than the spec definition.
+	if spec.IsChangedOnDisk(spec.Key.Path) || spec.IsChangedOnDisk(spec.Cert.Path) {
+		// This is necessary to essentially force cfssl to regenerate since it's not spec aware.
+		log.Infof("refreshing due to spec %s having a newer mtime than key or cert", spec.Path)
+		spec.ResetLifespan()
```

Comment:
Note- if you've not traced this function, I'd suggest doing so.  What I wired there was a hack and I don't really trust it to be comprehensive.

Reply:
The logic for checking mtimes makes sense to me. Are you worried about the usage of `spec.ResetLifespan()` ?

Reply:
moreso I was suggesting you trace that function since what I did in there is pretty horrible, and I'd appreciate a second set of eyes on it :)